adapters/Unity/UnityAdapter/GADMAdapterUnity.m

```diff
@@ -13,33 +13,42 @@
 // limitations under the License.
 
 #import "GADMAdapterUnity.h"
-
+#import "GADUnityError.h"
+#import "GADMAdapterUnityUtils.h"
 #import "GADMAdapterUnityBannerAd.h"
 #import "GADMAdapterUnityConstants.h"
-#import "GADMAdapterUnitySingleton.h"
-#import "GADMAdapterUnityUtils.h"
+#import "GADMUnityInterstitialAd.h"
 #import "GADMediationAdapterUnity.h"
 
-@interface GADMAdapterUnity () {
+@interface GADMAdapterUnity (){
   /// Connector from Google Mobile Ads SDK to receive ad configurations.
   __weak id<GADMAdNetworkConnector> _networkConnector;
-
+  
   /// Game ID of Unity Ads network.
   NSString *_gameID;
-
+  
   /// Placement ID of Unity Ads network.
   NSString *_placementID;
-
+  
   /// Unity Ads Banner wrapper
   GADMAdapterUnityBannerAd *_bannerAd;
+  
+  /// Unity Ads Interstitial Ad wrapper
+  GADMUnityInterstitialAd *_interstitialAd;
+  
+}
 
-  /// UUID for Unity instrument analysis
-  NSString *_uuid;
+@end
 
-  /// MetaData for storing Unity instrument analysis
-  UADSMetaData *_metaData;
+@interface GADMUnityInitializationDelegate : NSObject {
+  GADMediationAdapterSetUpCompletionBlock initCompletionBlock;
 }
 
+- (nonnull instancetype)initWithCompletionHandler:(GADMediationAdapterSetUpCompletionBlock)completionHandler;
+
+- (void)initializationComplete;
```

Comment:
Shouldn't need to define this API here. This is just Unity's UnityAdsInitializationDelegate API, which is defined by saying this class implements that delegate.